google-beta/resource_compute_instance_group_manager.go

```diff
@@ -686,7 +745,96 @@ func flattenAutoHealingPolicies(autoHealingPolicies []*computeBeta.InstanceGroup
 	return autoHealingPoliciesSchema
 }
 
+func flattenUpdatePolicy(updatePolicy *computeBeta.InstanceGroupManagerUpdatePolicy) []map[string]interface{} {
+	results := []map[string]interface{}{}
+	if updatePolicy != nil {
+		up := map[string]interface{}{}
+		if updatePolicy.MaxSurge != nil {
+			up["max_surge_fixed"] = updatePolicy.MaxSurge.Fixed
+			up["max_surge_percent"] = updatePolicy.MaxSurge.Percent
+		} else {
+			up["max_surge_fixed"] = 1
+			up["max_surge_percent"] = 0
+		}
+		if updatePolicy.MaxUnavailable != nil {
+			up["max_unavailable_fixed"] = updatePolicy.MaxUnavailable.Fixed
+			up["max_unavailable_percent"] = updatePolicy.MaxUnavailable.Percent
+		} else {
+			up["max_unavailable_fixed"] = 1
+			up["max_unavailable_percent"] = 0
+		}
+		up["min_ready_sec"] = updatePolicy.MinReadySec
+		up["minimal_action"] = updatePolicy.MinimalAction
+		up["type"] = updatePolicy.Type
+		results = append(results, up)
+	}
+	return results
+}
+
 func resourceInstanceGroupManagerStateImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(*Config)
 	d.Set("wait_for_instances", false)
+	id, err := parseInstanceGroupManagerId(d.Id())
+	if err != nil {
+		return nil, err
+	}
+	if id.Project == "" {
+		project, err := getProject(d, config)
+		if err != nil {
+			return nil, err
+		}
+		id.Project = project
+	}
+	if id.Zone == "" {
+		id.Zone, _ = getZone(d, config)
+		if id.Zone == "" {
+			region, err := getRegion(d, config)
+			if err != nil {
+				return nil, err
+			}
+
+			getInstanceGroupManager := func(zone string) (interface{}, error) {
+				return config.clientComputeBeta.InstanceGroupManagers.Get(id.Project, zone, id.Name).Do()
+			}
+			mgr, err := getZonalBetaResourceFromRegion(getInstanceGroupManager, region, config.clientComputeBeta, id.Project)
+			if err != nil {
+				return nil, err
+			}
+			if mgr != nil {
+				id.Zone = mgr.(*computeBeta.InstanceGroupManager).Zone
+			}
+		}
+	}
+	d.Set("project", id.Project)
+	d.Set("zone", id.Zone)
+	d.Set("name", id.Name)
 	return []*schema.ResourceData{d}, nil
 }
+
+type instanceGroupManagerId struct {
+	Project string
+	Zone    string
+	Name    string
+}
+
+func (i instanceGroupManagerId) terraformId() string {
+	return fmt.Sprintf("%s/%s/%s", i.Project, i.Zone, i.Name)
+}
+
+func parseInstanceGroupManagerId(id string) (*instanceGroupManagerId, error) {
```

Comment:
I think the pattern elsewhere for "parseStuffAndThings" ends up taking the resource obj and setting things like `project` and `zone` directly on `d`. I don't think there's enough utility exposed by this struct to keep it around. If you need something a more functional you could use multiple returns for project/zone/name.

Reply:
This is largely copying a pattern we've established elsewhere of using structs for multi-part IDs. If you grep the codebase for `terraformId`, you should see plenty of them pop up.